tests/integration/framework/process/daprd/options.go

```diff
@@ -136,6 +146,37 @@ func WithResourceFiles(files ...string) Option {
 	}
 }
 
+// WithSQLiteStateStore adds a SQLite state store component, which is also enabled as actor state store, and also returns a connection to the database.
+func WithSQLiteStateStore(t *testing.T) (Option, *sql.DB, error) {
+	// Create a SQLite database in the test's temporary directory
+	tmpDir := t.TempDir()
+	dbPath := filepath.Join(tmpDir, "test-data.db")
+	t.Logf("Storing SQLite database in %s", dbPath)
+
+	// Get an Option that adds the component
+	opt := WithResourceFiles(`
+apiVersion: dapr.io/v1alpha1
+kind: Component
+metadata:
+  name: mystore
+spec:
+  type: state.sqlite
+  version: v1
+  metadata:
+    - name: connectionString
+      value: 'file:` + dbPath + `'
+    - name: actorStateStore
+      value: 'true'
+`)
+
+	// Connect to the database
+	conn, err := sql.Open("sqlite", "file://"+dbPath+"?_txlock=immediate&_pragma=busy_timeout(5000)&_pragma=journal_mode(WAL)")
```

Comment:
Connection should be closed when test is cleaned up.

Also consider moving SQLite to a separate framework process. We should only ever be retuning `*Options` from option funcs, so this suggests this setup isn't quite right.

Reply:
No need to close, the connection is automatically closed when the object is deallocated. It's also a temporary file so doesn't matter anyways.

As for a separate process, SQLite is not a process that is spawned, so not sure I understand what you mean.

Reply:
We want to make this a separate process so that it itself can be passed to other options. The SQLite process would then expose a `DBFile() string` method which can be used by receiving Option funcs.

I'd rather we did close the connection for completeness. We will then know for sure all resources are cleaned up on test completion.

Reply:
To be clear here- an Integration framework process doesn't refer to a Linux process (although it can), it instead refers to a long running test dependency which has a Setup, Run, and exposes runtime config which can be used by other processes or the test itself. In this case, the SQLite will expose the db file created, and/or a connection to that file.